### DISCUSSION TITLE
🚀 FR Optimizations to file field loading in the cp

### DISCUSSION THREAD

**Discussion by robinsowell** [ExpressionEngine/ExpressionEngine#3383] [Feature Requests]
### Suggest an idea for ExpressionEngine

With large grids/file grids, the cp can get really slow just loading all the images. Not even on the EE end of things- the produced html can take a long time to load pulling in all the thumbs.

Case today- file grid with 250 rows.  The logic of his site is sound, but if you have 250 thumbs loading, the page in the cp is flat slow.

I did a test- made a static copy of the page to see how long it took

<img width="1260" alt="Screenshot 2023-05-11 at 12 42 10 PM" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/1181219/3c52906e-23df-4ee3-ba77-640e758dcdc7">

Then, I edited the static page to not load the thumbs
<img width="804" alt="Screenshot 2023-05-11 at 12 32 14 PM" src="https://github.com/ExpressionEngine/ExpressionEngine/assets/1181219/93d44394-abbd-45df-8f6a-c2150d6b00a6">

Big improvement.  Still slow, but way closer to acceptable. 

I'm wondering if we do something clever so all images in a file field pull in after page load so they don't slow down things.  Or lazy load, so only the ones showing load at first. It would have tiny but real benefits in general, and when we get into cases where there are lots of images, it would be a significant improvement.

**Comment by intoeetive** [comment]
Getting resolved in https://github.com/ExpressionEngine/ExpressionEngine/pull/3394
